task_processing/registry.py

```diff
@@ -0,0 +1,93 @@
+import importlib
+
+import pyrsistent
+
+from task_processing.interfaces import TaskExecutor
+
+
+class Registry(object):
+    """Registry for Task Processing plugins
+
+    This is the entrypoint to registering plugins (which know how to
+    actually run tasks) and creating instances of TaskExecutors. Note that
+    since most customers of this library will be importing this class into
+    their code, let's try to make it Python2/3 compatible
+
+    Typical use would be:
+
+        reg = Registry()
+        # registers 'myplugin'
+        reg.load_plugin('my.plugin')
+        # ... within the my.plugin.register_plugin method:
+        registry.add_task_executor('foobar', FooBarExecutor)
+
+        # register 'another'
+        reg.load_plugin('another.plugin')
+
+        # Later...
+        executor = reg.executor_from_config(
+            provider='foobar',
```

Comment:
hmm.. i'm not sure about plugin providing multiple executors. package providing multiple plugins - sure.

are you thinking about this:

    plugins/
      mesos/
        mesos_api_executor.py
        mesos_http_executor.py

or this:

    plugins/
      mesos_api/
        executor.py
      mesos_http/
        executor.py

?